src/cobs.cc

```diff
@@ -124,21 +124,25 @@ namespace cobs {
         uint8_t * dst_copy = dst_start;
         uint8_t * dst_offset = dst_copy++;
 
-        auto result = [&dst_copy, &dst_start](EncodeResult::Status status) -> EncodeResult {
-            return EncodeResult {
-                status,
+        auto bytes_produced = [&dst_copy, &dst_start]() -> size_t {
+            // Can cast because dst_copy >= dst_start.
+            return (size_t) (dst_copy - dst_start);
```

Comment:
Please use C++ style casts as they are more explicit with regards to your intention. http://en.cppreference.com/w/cpp/language/static_cast